core/src/main/java/org/wso2/msf4j/ServiceMethodInfo.java

```diff
@@ -28,12 +29,14 @@
 
     private final String methodName;
     private final Method method;
+    private Object[] args;
 
     private Map<String, Object> attributes = new ConcurrentHashMap<>();
 
-    public ServiceMethodInfo(String methodName, Method method) {
+    public ServiceMethodInfo(String methodName, Method method, Object[] args) {
         this.methodName = methodName;
         this.method = method;
+        this.args = Arrays.copyOf(args, args.length);;
```

Comment:
I had originally used without copying. But the problem was args[] has a "final" modifier but if we do not use array.copy(), internal objects can be modified by outside.